### PR TITLE
Update formatCurrency usage

### LINKED ISSUE #809: Move off deprecated formatCurrency function to new formatAmount
The `formatCurrency` function in the WooCommerce Admin currency module has been deprecated ([issue](https://github.com/woocommerce/woocommerce-admin/issues/4860) and [PR](https://github.com/woocommerce/woocommerce-admin/pull/4891/files)). We should figure out a way of moving onto the new function `formatAmount`.

Is it sufficient to wait until WooCommerce 4.3 falls outside out support policy and then switch over at that point, bumping the minimum version of WooCommerce?

### PR DESCRIPTION
**Pull request by vbelolapotkov**
Fixes #809 

#### Changes proposed in this Pull Request

* Use `currency.formatAmount` method instead of depricated `currency.formatCurrency` method.
* Bump `@woocommerce/currency` dependency to the latest version

#### Testing instructions

* Checks should be green.
* Install latest WC and check all pages where .formatCurrency was used before. Currencies should be properly displayed.

**Note: changelog entry is not needed because it's an internal change.**

-------------------

- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -81,7 +81,7 @@
     "@testing-library/react": "11.1.2",
     "@testing-library/user-event": "10.4.1",
     "@woocommerce/components": "5.1.2",
-    "@woocommerce/currency": "2.0.0",
+    "@woocommerce/currency": "^3.0.0",
```
Comment (kalessil) [inline package.json:84]: I'd suggest using "3.0.0" here, so deps bumping remaining under our control (according to how we setup renovate bot).

File: client/utils/currency/index.js [hunk 2]
```diff
@@ -52,12 +52,23 @@ export const formatCurrency = ( amount, currencyCode ) => {
...
+		return 'function' === typeof currency.formatAmount
+			? currency.formatAmount( amount )
+			: currency.formatCurrency( amount );
+	} catch ( err ) {
+		return composeFallbackCurrency( amount, currencyCode, isZeroDecimal );
 	}
-	return currency.formatCurrency( amount );
 };
+
+function composeFallbackCurrency( amount, currencyCode = '', isZeroDecimal ) {
```
Comment (kalessil) [inline client/utils/currency/index.js:67]: We don't need default value for `currencyCode` here, it's provided in all given invocations.

**Review by kalessil — APPROVED** [review]
Left two comments, but overall works and LGTM. Pre-approving.

PS: if I understand correctly deposit currency is not available yet, do we ahve a follow up ticket/PR for this?

File: package.json (hunk 1 above)
Comment (vbelolapotkov) [reply package.json]: fixed in https://github.com/Automattic/woocommerce-payments/pull/1236/commits/679b298a29d2b7471026557d35da795397b5b257

File: client/utils/currency/index.js (hunk 2 above)
Comment (vbelolapotkov) [reply client/utils/currency/index.js]: Good point, fixed in https://github.com/Automattic/woocommerce-payments/pull/1236/commits/4888871975ce299e66c1c9eb1085c2cba7b89fed

**Comment by vbelolapotkov** [comment]
> PS: if I understand correctly deposit currency is not available yet, do we ahve a follow up ticket/PR for this?

As discussed in slack this will be added once other store currencies are supported.

**Pull request merged**